app/src/full/java/io/homeassistant/companion/android/vehicle/MainVehicleScreen.kt

```diff
@@ -67,30 +125,81 @@ class MainVehicleScreen(
                             if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()
                         }
                     }
+            val icon = Entity(
+                "$domain.ha_android_placeholder",
+                "",
+                mapOf<Any, Any>(),
+                Calendar.getInstance(),
+                Calendar.getInstance(),
+                null
+            ).getIcon(carContext)
+
             listBuilder.addItem(
-                Row.Builder()
+                Row.Builder().apply {
+                    if (icon != null) {
+                        setImage(
+                            CarIcon.Builder(
+                                IconicsDrawable(carContext, icon)
+                                    .apply {
+                                        sizeDp = 48
+                                    }.toAndroidIconCompat()
+                            )
+                                .setTint(CarColor.DEFAULT)
+                                .build()
+                        )
+                    }
+                }
                     .setTitle(friendlyDomain)
                     .setOnClickListener {
-                        Log.i(TAG, "$domain clicked")
+                        Log.i(TAG, "Domain:$domain clicked")
                         screenManager.push(
                             EntityGridVehicleScreen(
                                 carContext,
                                 integrationRepository,
                                 friendlyDomain,
-                                entities.filter { it.value.domain == domain }.toMutableMap()
+                                allEntities.map { it.values.filter { entity -> entity.domain == domain } }
                             )
                         )
                     }
                     .build()
             )
         }
 
-        // TODO: Add row for zones so we can start navigation?
+        listBuilder.addItem(
+            Row.Builder()
+                .setImage(
+                    CarIcon.Builder(
+                        IconicsDrawable(
+                            carContext,
+                            CommunityMaterial.Icon3.cmd_map_outline
+                        ).toAndroidIconCompat()
```

Comment:
Missed this one when increasing the sizes, it is still blurry

Reply:
Ooops..... Ty.